frontend/src/Components/CartCard.js

```diff
@@ -0,0 +1,97 @@
+import { Link, useNavigate } from "react-router-dom";
+import { useCart } from "../Contexts/cartContext";
+import { useAuth } from "../Contexts/authContext";
+import { useEffect, useState } from "react";
+import { useToast } from "../Contexts/toastContext";
+import "./Styles/cartCard.scss";
+import { useWishlist } from "../Contexts/wishlistContext";
+
+const CartCard = ({
+  productDetails: { product, name, price, image, description, qty },
+}) => {
+  const [qtyCounter, setQtyCounter] = useState(qty);
+  const { addToCart, removeFromCart } = useCart();
+  const { addToWishlist } = useWishlist();
+  const { addToast } = useToast();
+
+  const handleAddToWishlist = (id) => {
+    addToWishlist(id);
+    setTimeout(() => {
+      removeFromCart(id);
+    }, 0);
+  };
+
+  const handleQuantity = (action) => {
+    if (action === "DECREMENT" && qtyCounter > 1) {
+      setQtyCounter(qtyCounter - 1);
+    } else if (action === "INCREMENT") {
+      setQtyCounter(qtyCounter + 1);
+    }
+
+    return null;
+  };
+
+  const handleRemoveItem = () => {
+    addToast({ type: "success", message: "Item removed from cart" });
+    removeFromCart(product);
+  };
+
+  useEffect(() => {
+    addToCart(product, qtyCounter);
```

Comment:
I think this is the wrong way to call addToCart. As it is not a component functionality. It will be called on initial Render of the component as well. this functionality is dependent on user. When either user clicks add to cart or increment quantity.